libkbfs/folder_branch_ops.go

```diff
@@ -1907,6 +1907,34 @@ func (bps *blockPutState) mergeOtherBps(other *blockPutState) {
 	bps.blockStates = append(bps.blockStates, other.blockStates...)
 }
 
+func (bps *blockPutState) removeOtherBps(other *blockPutState) {
+	if len(other.blockStates) == 0 {
+		return
+	}
+
+	otherPtrs := make(map[BlockPointer]bool, len(other.blockStates))
+	for _, bs := range other.blockStates {
+		otherPtrs[bs.blockPtr] = true
+	}
+
+	// Assume that `other` is a subset of `bps` when initializing the
+	// slice length.
+	newLen := len(bps.blockStates) - len(other.blockStates)
+	if newLen <= 0 {
+		newLen = 1
```

Comment:
Although at the beginning we return 0 if `len(other.blockStates) == 0`, it's still possible to have an empty `newBlockStates` later right? If `cap` matters why not use `0` here?